apps/model/model.js

```diff
@@ -671,6 +686,16 @@ function uploadModel() {
           return;
         }
 
+        try {
+          if (checkNameDuplicate(_name.value)) {
+            throw new Error('Model name repeated');
+          }
+        } catch (e) {
+          status.innerHTML = 'Model with the same name already exists. Please choose a new name';
+          status.classList.remove('blink');
```

Comment:
Needs to return form here else it will still save the model.

Reply:
Thanks for pointing out. I have fixed it.